### PR TITLE
[Woo POS][Non-Simple Products] Add variation to cart

### LINKED ISSUE #12845: [POS][Non-Simple Product type] Add variation to cart when clicked
The goal of this task is to add the variation into the cart when clicked.

Image

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #12845 

### Description

This PR adds variation to the cart when clicked from the variations list screen. 

### Testing information

Try adding different variations of a variable product to the cart and ensure they all get added to the cart with correct name and variation attribute.

### The tests that have been performed

Adding different variations to the cart in both dark and light mode.

1. Navigate to the POS mode (more menu -> POS)
2. Select any variable product
3. Ensure you navigate to the variations list screen
4. Click on any variation
5. Ensure that variation gets added to the cart

### Images/gif

https://github.com/user-attachments/assets/847c670c-8e43-46a7-afa5-01411e72fa75

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/244">21.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModel.kt [hunk 1]
```diff
@@ -112,43 +116,62 @@ class WooPosCartViewModel @Inject constructor(
         viewModelScope.launch {
             parentToChildrenEventReceiver.events.collect { event ->
                 when (event) {
-                    is ParentToChildrenEvent.BackFromCheckoutToCartClicked -> {
-                        _state.value = _state.value.copy(cartStatus = EDITABLE)
-                    }
+                    is ParentToChildrenEvent.BackFromCheckoutToCartClicked -> handleBackFromCheckoutToCartClicked()
+                    is ParentToChildrenEvent.ItemClickedInProductSelector -> handleItemClickedInProductSelector(event)
+                    is ParentToChildrenEvent.OrderSuccessfullyPaid -> handleOrderSuccessfullyPaid()
+                    is ParentToChildrenEvent.CheckoutClicked -> handleCheckoutClicked()
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModel.kt]: 💡 np.: It's a bit confusing to redirect to an empty function (handleCheckoutClicked()). How about this?
```suggestion
                    is ParentToChildrenEvent.CheckoutClicked -> Unit
```

**Review by samiuelson — APPROVED** [review]
LGTM

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModel.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModel.kt]: Done: https://github.com/woocommerce/woocommerce-android/pull/12926/commits/1daccc8d0e3d975ab25ef5eda4fc0940c6d7d2aa

**Pull request merged**